context/context.go

```diff
@@ -0,0 +1,67 @@
+package context
+
+import (
+	gocontext "context"
+	"fmt"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Context interface {
+	Errorf(err error)
+	StartSpan(tracer trace.Tracer, spanName string) trace.Span
+	SetSpanAttributes(key string, val any)
+}
+
+func NewContext() Context {
+	return &context{}
+}
+
+// context implements Context
+type context struct {
+	gocontext.Context
+}
```

Comment:
Should we have this as the only context and every other context that is derived of it or uses it can add other fields but not any context, will that help ?

Reply:
The duty context should be extending this context, not gocontext